plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraTable.java

```diff
@@ -0,0 +1,176 @@
+ * limitations under the License.
+ */
+package io.trino.plugin.cassandra;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.testing.sql.SqlExecutor;
+
+import java.security.SecureRandom;
+import java.util.List;
+import java.util.function.Function;
+
+import static com.google.common.base.Verify.verify;
+import static java.lang.Character.MAX_RADIX;
+import static java.lang.Math.abs;
+import static java.lang.Math.min;
+import static java.lang.String.format;
+import static java.lang.String.join;
+import static java.util.stream.Collectors.joining;
+
+public class TestCassandraTable
+        implements AutoCloseable
+{
+    private static final SecureRandom random = new SecureRandom();
+    // The suffix needs to be long enough to "prevent" collisions in practice. The length of 5 was proven not to be long enough
+    private static final int RANDOM_SUFFIX_LENGTH = 10;
+
+    private final SqlExecutor sqlExecutor;
+    private final String keyspace;
+    private final String tableName;
+
+    // Make sure the `namePrefix` in the following constructors does not exceed 38 characters,
+    // because Cassandra table names are limited to 48 characters in size, so we can only use
+    // a maximum of 38 characters without the suffix.
+    // https://cassandra.apache.org/doc/trunk/cassandra/cql/ddl.html#common-definitions
+    //
+    // NOTE: "38 characters" may be changed in the future, please refer to
+    // `48 - RANDOM_SUFFIX_LENGTH` for the exact value.
+
+    public TestCassandraTable(
+            SqlExecutor sqlExecutor,
+            CassandraServer server,
+            String keyspace,
+            String namePrefix,
+            List<ColumnDefinition> columnDefinitions,
+            List<String> rowsToInsert)
+    {
+        this.sqlExecutor = sqlExecutor;
+        this.keyspace = keyspace;
+        this.tableName = namePrefix + randomTableSuffix();
+        sqlExecutor.execute(format("CREATE TABLE %s.%s %s", keyspace, tableName, tableDefinition(columnDefinitions)));
+        try {
+            String columns = columnDefinitions.stream()
+                    .map(columnDefinition -> columnDefinition.name)
+                    .collect(joining(", "));
+            for (String row : rowsToInsert) {
+                // some databases do not support multi value insert statement
+                sqlExecutor.execute(format("INSERT INTO %s.%s (%s) VALUES (%s)", keyspace, tableName, columns, row));
+            }
+
+            // Ensure that the currently created table is visible to other sessions (e.g. the session used in DistributedQueryRunner)
+            server.refreshSizeEstimates(keyspace, tableName);
+        }
+        catch (Exception e) {
+            try (TestCassandraTable ignored = this) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    public String getTableName()
+    {
+        return tableName;
+    }
+
+    public String getTableNameInKeyspace()
```

Comment:
Remove this method and let `getTableName()` returns `format("%s.%s", keyspace, tableName)`. 

Then, we can remove redundant `assertContainsEventually()` in `testEmptyTimestampClusteringKey` & `testNullAndEmptyTimestamp`.

Reply:
I think the tests in [this PR](#11082) can be replaced with `TestCassandraTable` to avoid writing repetitive and boring `CREATE TABLE`, `INSERT INTO`, `DROP TABLE` statements.

https://github.com/trinodb/trino/pull/11082/files#diff-ba828445c7192ac4b49069fa8ff98608e64e596b0d29a21fac09f50469316deaR276-R290

Reply:
I'll submit a new PR if you think it's workable (let's try to keep the current PR simple).